src/files.rs

```diff
@@ -19,16 +18,37 @@ pub fn read_file(path: &Path) -> Fallible<Vec<u8>> {
     std::fs::read(&path).map_err(|err| format!("{}: {}", err, path.display()).into())
 }
 
+fn files_differ(old: &Path, new: &Path) -> bool {
+    let old_content = match read_file(old) {
+        Ok(c) => c,
+        Err(_) => return true,
+    };
+    let new_content = match read_file(new) {
+        Ok(c) => c,
+        Err(_) => return true,
+    };
+    old_content != new_content
+}
+
+pub fn file_differs(file: &Path, content: &str) -> bool {
+    let other_content = match read_file_string(file) {
```

Comment:
I think this would be a good candidate for `map_or`, e.g.
```rust
read_file_string(file).map_or(true, |content| content != new_content)
```
and similarly above
```rust
read_file(old).and_then(|old|  read_file(new).map(|new| old != new)).unwrap_or(true)
```